go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/aquasecurity/libbpfgo
 
-go 1.21
+go 1.22.0
```

Comment:
Is it necessary to bump the minimum Go version?

This has a knockdown effect on everything that depends on `libbpfgo`:

```diff
$ git diff go.mod
diff --git a/go.mod b/go.mod
index dd5b16f..f1fece8 100644
--- a/go.mod
+++ b/go.mod
@@ -1,10 +1,12 @@
 module github.com/cloudflare/ebpf_exporter/v2

-go 1.21
+go 1.22.0
+
+toolchain go1.22.3

 require (
        github.com/alecthomas/kingpin/v2 v2.4.0
-       github.com/aquasecurity/libbpfgo v0.7.0-libbpf-1.4
+       github.com/aquasecurity/libbpfgo v0.7.0-libbpf-1.4.0.20240524222422-daeb5d0059fe
        github.com/coreos/go-systemd v0.0.0-20191104093116-d3cd4ed1dbcf
        github.com/elastic/go-perf v0.0.0-20191212140718-9c656876f595
        github.com/iovisor/gobpf v0.2.0
```

I think it also forces Go upgrade from v1.22.2 to v1.22.3:

```
go: downloading go1.22.3 (linux/arm64)
```

Reply:
> Is it necessary to bump the minimum Go version?
> 
> This has a knockdown effect on everything that depends on `libbpfgo`:

Yes, it really enforces consumers to align versions.

There are improvements in 1.22 that can be useful for all of us, including performance ones: https://tip.golang.org/doc/go1.22

@bobrik can't you bump it at your side?

> 
> I think it also forces Go upgrade from v1.22.2 to v1.22.3:
> 
> ```
> go: downloading go1.22.3 (linux/arm64)
> ```

Don't worry about this, it's the toolchain requirement, it won't update your system go. It downloads and uses for the current project.

Reply:
You are not supposed to dictate what go version is necessary for the consumers unless it's actually necessary:

* https://go.dev/doc/modules/gomod-ref#go

No matter how good the latest Go version is, if it's not a requirement, you are not supposed to force it. You set the minimum supported Go version instead. Go supports two major releases at any time and it seems reasonable to do at least the same thing.

People with patched go versions might not have an ability to upgrade to the latest release.

I don't think there's a good reason to set `toolchain` for a library ever.

Reply:
You're absolutely right. I'm going to revert (soon) this PR by splitting the fix and other chores into their own PRs. 👍🏼

Reply:
Thanks, I appreciate it.